src/components/CardMetadataDisplay.tsx

```diff
@@ -95,16 +77,39 @@ const CardMetadataDisplay = (props: CardMetadataDisplayProps) => {
             </Show>
             <Show when={props.signedInUserId == props.viewingUserId}>
               <button
+                title="Delete"
                 classList={{
                   "h-fit text-red-700 dark:text-red-400": true,
                   "animate-pulse": deleting(),
                 }}
-                onClick={() => deleteCard()}
+                onClick={() => {
+                  if (props.signedInUserId !== props.viewingUserId) return;
+
+                  props.setOnDelete(() => {
```

Comment:
you shouldn't access props directly within jsx. Refer to: https://github.com/arguflow/vault/pull/59/commits/590534033002e7b2440019a93875786a2b6a3e32#diff-41fb5f4b11b912504e1619a4e1e82da4a7f38059e45c678103c55859960c3b2a to see how nick fixed it